xbmc/addons/kodi-addon-dev-kit/src/api2/kodi-addon-sharedlibrary/InterProcess.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ *      Copyright (C) 2016 Team KODI
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with KODI; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "InterProcess.h"
+
+#include <stdio.h>
+
+#ifdef _WIN32                   // windows
+  #ifndef _SSIZE_T_DEFINED
+    typedef intptr_t ssize_t;
+    #define _SSIZE_T_DEFINED
+  #endif // !_SSIZE_T_DEFINED
+  #include "dlfcn-win32.h"
+#else
+  #include <dlfcn.h>            // linux+osx
+#endif
+
+extern "C"
+{
+
+CKODIAddon_InterProcess g_interProcess;
```

Comment:
no globals, no singletons (which are globals too)
The object needs to be own by another object. The top level object is CApplication and we want to startup/shutdown the app in a controlled manner. Checkout ServiceManager and ServiceBroker